framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -178,11 +344,21 @@ class BlockSynchronizationMechanism {
 			height: peers[0].height,
 		};
 
-		if (!ForkChoiceRule.isDifferentChain(this.blocks.lastBlock, peersTip)) {
+		const forkStatus = await this.processorModule.forkStatus.run({
+			block: peersTip,
```

Comment:
peersTip needs to have `version` to select the correct block_processor.
Also, I think `run` is the method for pipeline, and `processorModule.forkStatus` should be fine (although it should be exposed from `processor/processor.js`

Reply:
Done :)